### PR TITLE
Fix selection with end home

### PR DESCRIPTION
**Pull request by eymar**
Window and Linux will select only the line where the cursor is.
MacOS will select till the end/start of the text.

### CODE REVIEW

File: compose/foundation/foundation/src/desktopMain/kotlin/androidx/compose/foundation/text/KeyMapping.desktop.kt [hunk 1]
```diff
@@ -110,8 +110,8 @@ internal actual val platformDefaultKeyMapping: KeyMapping =
 
                         event.isShiftPressed ->
                             when (event.key) {
-                                MappedKeys.MoveHome -> KeyCommand.SELECT_HOME
-                                MappedKeys.MoveEnd -> KeyCommand.SELECT_END
+                                MappedKeys.MoveHome -> KeyCommand.SELECT_LINE_START
```
Comment (igordmn) [inline compose/foundation/foundation/src/desktopMain/kotlin/androidx/compose/foundation/text/KeyMapping.desktop.kt]: Should we change it in `defaultKeyMapping` as well? I don't have a keyboard with Home button right now, so I can't verify it on Windows and Android.

Comment (igordmn) [reply compose/foundation/foundation/src/desktopMain/kotlin/androidx/compose/foundation/text/KeyMapping.desktop.kt]: Actually, I can press Fn + Left Arrow ).

On macOs we select the whole text, not the whole line. Tested in TextEdit and in Pages

Comment (eymar) [reply compose/foundation/foundation/src/desktopMain/kotlin/androidx/compose/foundation/text/KeyMapping.desktop.kt]: I tested in google docs (mac os), and there they select the line only. And in IDEA on mac os, it selects only the line.

What do you think, Should we make this behaviour dependant on a platform?

Comment (igordmn) [reply compose/foundation/foundation/src/desktopMain/kotlin/androidx/compose/foundation/text/KeyMapping.desktop.kt]: Checked on Windows - when we press Home/End, we move to the start/end of the line. If we press Shift + Home/End - we select part of the line

Comment (igordmn) [reply compose/foundation/foundation/src/desktopMain/kotlin/androidx/compose/foundation/text/KeyMapping.desktop.kt]: > What do you think, Should we make this behaviour dependant on a platform?

I think, key behaviour should mimic the native behaviour.

Google Docs and IDEA probably not best role models :).

Best role models are native apps, then, probably Chromium. In TextArea on macOs we have the native behaviour.

Comment (igordmn) [reply compose/foundation/foundation/src/desktopMain/kotlin/androidx/compose/foundation/text/KeyMapping.desktop.kt]: The current non-spoken strategy was to mimic feel on native platform, but not mimic look.

Comment (eymar) [reply compose/foundation/foundation/src/desktopMain/kotlin/androidx/compose/foundation/text/KeyMapping.desktop.kt]: Updated the PR. 
Window and Linux will select only the line where the cursor is.
MacOS will select till the end/start of the text.

The tests use reflection to mock the top level field `platformDefaultKeyMapping`.

**Review by igordmn — APPROVED** [review]

**Pull request merged**